app/models/finance/billing_strategy.rb

```diff
@@ -166,7 +166,7 @@ def create_invoice(opts = {})
   end
 
   def create_invoice_counter(period)
-    invoice_prefix = (billing_monthly? ? period : period.begin.year).to_param
+    invoice_prefix = period.to_param[0..(billing_monthly? ? 6 : 3)]
```

Comment:
I think it is because you serialize it as a date?

Reply:
Strictly speaking, the invoice prefix is just a string, and that indeed failed in a few tests I ran. Anyway, it's a bug introduced when we merged the invoice counter. So I changed. Now it works if you pass either a `String` or a `Month`.